src/components/NumericStepper/NumericStepper.jsx

```diff
@@ -0,0 +1,330 @@
+import React, {PropTypes, Component} from 'react';
+import {themr} from 'react-css-themr';
+import {DEBOUNCE} from 'dx-util/src/function/function.js';
+import {PURE} from 'dx-util/src/react/pure';
+import Input from '../Input/Input.jsx';
+import ButtonIcon from '../ButtonIcon/ButtonIcon.jsx';
+import Button from '../Button/Button.jsx';
+import classnames from 'classnames';
+
+export const NUMERIC_STEPPER = Symbol('NumericStepper');
+
+const BUTTON_TYPE = {
+	UP: 'up',
+	DOWN: 'down'
+};
+
+const KEYCODE = {
+	UP: 38,
+	DOWN: 40
+};
+
+@PURE
+@themr(NUMERIC_STEPPER)
+export default class NumericStepper extends Component {
+
+	static propTypes = {
+		...Input.propTypes,
+		step: React.PropTypes.number,
+		min: React.PropTypes.number,
+		max: React.PropTypes.number,
+		formatter: React.PropTypes.func,
+		ButtonComponent: React.PropTypes.func,
+		InputComponent: React.PropTypes.func,
+
+		repeatIntervar: React.PropTypes.number,
+		repeatDelay: React.PropTypes.number,
+
+		upIconName: React.PropTypes.string,
+		downIconName: React.PropTypes.string,
+
+		theme: React.PropTypes.shape({
+			container: React.PropTypes.string,
+			container__input: React.PropTypes.string,
+			container__input_isCorrected: React.PropTypes.string,
+			container__buttons: React.PropTypes.string,
+			container__button: React.PropTypes.string,
+			container__button__icon: React.PropTypes.string,
+			container__button_up: React.PropTypes.string,
+			container__button_down: React.PropTypes.string
+		})
+	};
+
+	constructor(props) {
+		super(props);
+
+		this.state = {
+			value: props.value,
+			isCorrected: false,
+			isFocused: false
+		};
+	}
+
+	static defaultProps = {
+		step: 1,
+		max: Infinity,
+		min: 0,
+		value: 1,
+		pattern: /^-?[0-9]\d*(\.\d+)?$/,
+		isDisabled: false,
+		InputComponent: Input,
+		ButtonComponent: ButtonIcon
+	};
+
+	setValue(newValue) {
+		const {value} = this.state;
+		const {onChange} = this.props;
+
+		const correctedValue = this.getCorrectedValue(newValue);
+
+		const isChanged = correctedValue !== value;
+		const isCorrected = correctedValue !== newValue;
+
+		if (isChanged) {
+			this.setState({
+				value: correctedValue,
+				isCorrected
+			});
+			onChange && onChange(correctedValue);
+		}
+
+		if (!isChanged && isCorrected) {
+			this.setState({
+				isCorrected
+			});
+		}
+	}
+
+	getCorrectedValue(value) {
+		const {min, max} = this.props;
+		let newValue = parseFloat(value);
+		if (newValue < min) {
+			newValue = min;
+		}
+		if (newValue > max) {
+			newValue = max;
+		}
+		return newValue;
+	}
+
+	step(n) {
+		const {value} = this.state;
+		const {step} = this.props;
+		const newValue = (value || 0) + (step * n);
+		this.setValue(newValue);
+	}
+
+	increase = () => {
+		this.step(1);
+	}
+
+	decrease = () => {
+		this.step(-1);
+	}
+
+	componentWillReceiveProps(newProps) {
+		this.setValue(newProps.value);
+	}
+
+	componentWillUpdate(newProps, newState) {
+		if (newState.isCorrected) {
```

Comment:
if you get called multiple times here these timeouts can potentially end with a race condition